### DISCUSSION TITLE
Beginner's problem adding new device

### DISCUSSION THREAD

**Discussion by jl-net** [olikraus/u8g2#1401] [Q&A]
Dear All,
I am trying to add a new device (Display Elektronik DEM240128F) to the library so I can use it. The controller ST7586S  is already supported, therefore I hope I can get it run quite well. But I am struggeling in some problems and need some beginner's help:
1. What kind of tool chain/compiler/target has to be used to run "make codebuild"? AVR-GCC is not the solution because error.h is not existing and even if I "borrow" this one from somewhere else I get some more compiler and linker errors.
2. Is it possible (for testing purposes) to just change the code in one of the existing files and use the library without doing the "codebuild" run? This would speed up development.

Thank you for your support!

Jürgen

**Comment by olikraus** [comment]
make codebuild will run under any linux environment and probably also within mingw on Windows.
And of course yes, you can modify any existing device.
